system/extensions/esp32/wifi.toit

```diff
@@ -152,7 +152,13 @@ class WifiServiceProvider extends NetworkServiceProviderBase:
 
   scan config/Map -> List:
     if state_.module:
-      throw "wifi already connected or established"
+      if state_.module is WifiModule:
+        channels := config.get wifi.CONFIG-SCAN-CHANNELS
+        passive := config.get wifi.CONFIG-SCAN-PASSIVE
+        period := config.get wifi.CONFIG-SCAN-PERIOD
+        return (state_.module as WifiModule).scan channels passive period
```

Comment:
I think we need to make it so you can't connect while scanning, but you can scan while connected.

Reply:
done.